source/server/listener_impl.h

```diff
@@ -216,8 +228,35 @@ class ListenerImpl final : public Network::ListenerConfig,
   ListenerImpl(const envoy::config::listener::v3::Listener& config, const std::string& version_info,
                ListenerManagerImpl& parent, const std::string& name, bool added_via_api,
                bool workers_started, uint64_t hash, uint32_t concurrency);
+  /**
+   * Create a new listener from an existing listener and the new config message. This should be
+   * called only if the new config differs among filter chains.
+   */
```

Comment:
Can we just modify the existing listener in place? Would this help us reuse the same tag? If we have to create a new listener implementation can we figure out how to use the same tag per our previous conversation on this topic?